src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -63,17 +91,30 @@ def __init__(
         @param password: the password of the nso account
         @cloud_detection_model_path: optional location of a .sav of a cloud detection model
         """
-
-        self.path_to_geojson = correct_file_path(path_to_geojson)
-        # Name needs to be included in the geojson name.
-        self.region_name = path_to_geojson.split("/")[
-            len(path_to_geojson.split("/")) - 1
-        ].split(".")[0]
+        if path_to_geojson is not None:
+            self.path_to_geojson = correct_file_path(path_to_geojson)
+            # Name needs to be included in the geojson name.
+            self.region_name = path_to_geojson.split("/")[
+                len(path_to_geojson.split("/")) - 1
+            ].split(".")[0]
+
+        elif coordinates is not None:
+            self.region_name = "fill_region"
+
+        elif previous_path_to_geojson is not None and coordinates is not None:
```

Comment:
This elif statement will never be reached, because: if `coordinates is not None`, the previous elif statement is true and therefore it stops moving through the list of elif statements.

Reply:
correct